lib/Dialect/Emit/EmitOps.cpp

```diff
@@ -47,6 +47,18 @@ void FileOp::build(OpBuilder &builder, OperationState &result,
     bodyCtor();
 }
 
+//===----------------------------------------------------------------------===//
+// RefOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult RefOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto target = getTargetAttr();
+  auto *op = symbolTable.lookupNearestSymbolFrom(getOperation(), target);
```

Comment:
Can we reject unsupported op types in the verifier?

Reply:
The `emit` dialect can't link to the other dialects to reference the ops for `isa` AFAIK. We should introduce a trait that makes ops emittable, but that is a bit overkill at the moment.

Reply:
On second thought, I added the interface.